server/middleware/role-assign.js

```diff
@@ -0,0 +1,26 @@
+const jwt = require('jsonwebtoken');
+const config = require('../config');
+const logger = require('../logger');
+
+/**
+ * Assigns a role to the user in the request object
+ */
+module.exports = (req, res, next) => {
+  const token = req.headers.authorization.split(' ')[1];
+
+  // decode the token using a secret key-phrase
+  return jwt.verify(token, config.jwtSecret, (err, decoded) => {
+    const isAdmin = decoded.admin;
+
```

Comment:
You have already extracted the token in the auth-check middle ware, then why do it again here. Just pass the token[or admin enum value] in the req object from the auth middeleware to the role assign middleware.

Reply:
That is a good point

Reply:
Yeah, but I thought it would be better have to have a separate middleware for authorization(assigning a role to the user of the request) and the `auth-check` middleware will only perform authentication.

Reply:
You didnt get my point may be. Let the middlewares for authentication and authorisation be  differenet , bt in the auth check itself pass the token in the req object. Why redundantly perform the same action. I will suggest change this and amend the comment